include/marl/scheduler.h

```diff
@@ -50,8 +51,33 @@ class Scheduler {
  public:
   using TimePoint = std::chrono::system_clock::time_point;
   using Predicate = std::function<bool()>;
+  using ThreadInitializer = std::function<void(int workerId)>;
+
+  // Config holds scheduler configuration settings that can be passed to the
+  // Scheduler constructor.
+  struct Config {
+    // Per-worker-thread settings.
+    struct WorkerThread {
+      int count = 0;
+      ThreadInitializer initializer;
+    };
+    WorkerThread workerThread;
 
-  Scheduler(Allocator* allocator = Allocator::Default);
+    // Memory allocator to use for the scheduler and internal allocations.
+    Allocator* allocator = Allocator::Default;
+
+    // AllCores returns a Config with a worker thread for each of the logical
+    // cpus available to the process.
+    static Config AllCores();
```

Comment:
Should this be 'allCores()' (lower-case 'a')? Or do static functions start with upper-case?

Reply:
Good spot. Yes it should have. I should invest in setting up a linter.

Reply:
Done.